pkg/server/systemconfigwatcher/systemconfigwatchertest/test_system_config_watcher.go

```diff
@@ -118,18 +119,37 @@ func runTest(
 	waitForEqual(t)
 }
 
-func getSystemConfig(
+func getSystemDescriptorAndZonesSpans(
 	ctx context.Context, t *testing.T, codec keys.SQLCodec, kvDB *kv.DB,
 ) []roachpb.KeyValue {
 	var ba roachpb.BatchRequest
-	ba.Add(roachpb.NewScan(
-		append(codec.TenantPrefix(), keys.SystemConfigSpan.Key...),
-		append(codec.TenantPrefix(), keys.SystemConfigSpan.EndKey...),
-		false, // forUpdate
-	))
+	ba.Add(
+		roachpb.NewScan(
+			append(codec.TenantPrefix(), keys.SystemDescriptorTableSpan.Key...),
+			append(codec.TenantPrefix(), keys.SystemDescriptorTableSpan.EndKey...),
+			false, // forUpdate
+		),
+	)
 	br, pErr := kvDB.NonTransactionalSender().Send(ctx, ba)
 	require.NoError(t, pErr.GoError())
-	return br.Responses[0].GetScan().Rows
+
+	rows1 := br.Responses[0].GetScan().Rows
+
+	ba.Reset()
+
+	ba.Add(
+		roachpb.NewScan(
+			append(codec.TenantPrefix(), keys.SystemZonesTableSpan.Key...),
+			append(codec.TenantPrefix(), keys.SystemZonesTableSpan.EndKey...),
+			false, // forUpdate
+		),
+	)
+	br, pErr = kvDB.NonTransactionalSender().Send(ctx, ba)
+	require.NoError(t, pErr.GoError())
+
+	rows2 := br.Responses[0].GetScan().Rows
+
+	return append(rows1, rows2...)
```

Comment:
make a helper

Reply:
+1, a helper to construct this `row` thing for each span would help. Would also let you avoid resetting a batch request for re-use.